packages/investment-pool/contracts/InvestmentPool.sol

```diff
@@ -671,6 +689,8 @@ contract InvestmentPool is IInitializableInvestmentPool, SuperAppBase, Context,
             revert InvestmentPool__AlreadyStreamingForMilestone(_milestoneId);
         if (milestone.paid) revert InvestmentPool__AlreadyPaidForMilestone(_milestoneId);
 
+        _ifNeededUpdateMemInvestmentValue(_milestoneId);
```

Comment:
Are we sure that this is the only place that this check is required?
As far as I remember - `getTotalMilestoneTokenAllocation()` was used in multiple places and performed that check in several different scenarios.

Perhaps it's not possible for the contract to reach such state that this check is necessary in multiple places, but we need to make sure that it behaves as we expect.